pkg/specgen/generate/oci.go

```diff
@@ -165,7 +165,7 @@ func SpecGenToOCI(ctx context.Context, s *specgen.SpecGenerator, rt *libpod.Runt
 			inUserNS = true
 		}
 	}
-	if inUserNS && s.NetNS.IsHost() {
+	if inUserNS && !(s.NetNS.NSMode == specgen.NoNetwork || s.NetNS.NSMode == specgen.Private) {
```

Comment:
I am not sure this is correct?  And I think we should be using the functions in namespaces?  I am not sure if all cases are handled.

NSMode == "default" 
NSMode == ""
are the same as
NSMode == "private"

Reply:
No, they're not. We explicitly do not want to trigger this with slirp4netns networking, and both of those are trigger slirp.

Reply:
If `private` also triggers slirp, then we'll need to explicitly disable it as well.

Reply:
I'm going to revert back to NoNetwork. It's guaranteed to be safe. We can loosen this later if we need to.

Reply:
So we need to have a check added to indicate whether or not we are using slirp4netns networking.

Reply:
I'm actually not sure if CNI + user namespaces is safe either. It's probably better to leave disabled.